cmd/skopeo/copy.go

```diff
@@ -102,22 +103,36 @@ See skopeo(1) section "IMAGE NAMES" for the expected format
 
 // parseMultiArch parses the list processing selection
 // It returns the copy.ImageListSelection to use with image.Copy option
-func parseMultiArch(multiArch string) (copy.ImageListSelection, error) {
+func parseMultiArch(multiArch string) (copy.ImageListSelection, []manifest.Schema2PlatformSpec, error) {
 	switch multiArch {
 	case "system":
-		return copy.CopySystemImage, nil
+		return copy.CopySystemImage, nil, nil
 	case "all":
-		return copy.CopyAllImages, nil
+		return copy.CopyAllImages, nil, nil
 	// There is no CopyNoImages value in copy.ImageListSelection, but because we
 	// don't provide an option to select a set of images to copy, we can use
 	// CopySpecificImages.
-	case "index-only":
-		return copy.CopySpecificImages, nil
-	// We don't expose CopySpecificImages other than index-only above, because
-	// we currently don't provide an option to choose the images to copy. That
-	// could be added in the future.
 	default:
-		return copy.CopySystemImage, fmt.Errorf("unknown multi-arch option %q. Choose one of the supported options: 'system', 'all', or 'index-only'", multiArch)
+		platformsArr := strings.Split(multiArch, ",")
+		platforms := []manifest.Schema2PlatformSpec{}
+		for _, platform := range platformsArr {
+			parts := strings.SplitN(platform, "/", 2)
+			if len(parts) != 2 {
+				return copy.CopySystemImage, nil, fmt.Errorf("invalid platform format %q: expected os/arch", platform)
+			}
+
+			os := parts[0]
+			arch := parts[1]
+			platforms = append(platforms, manifest.Schema2PlatformSpec{OS: os, Architecture: arch})
+		}
+		return copy.CopySpecificImages, platforms, nil
+		// case "index-only":
+		// 	return copy.CopySpecificImages, nil
+		// // We don't expose CopySpecificImages other than index-only above, because
+		// // we currently don't provide an option to choose the images to copy. That
+		// // could be added in the future.
+		// default:
+		// 	return copy.CopySystemImage, fmt.Errorf("unknown multi-arch option %q. Choose one of the supported options: 'system', 'all', or 'index-only'", multiArch)
```

Comment:
Ultimately, code should be removed, unless it is relevant for future readers. Git preserves the history anyway.